### PR TITLE
TNO-670 Fix content redux hook

### PR DESCRIPTION
**Pull request by Fosol**
The redux state in the list view wasn't being updated reliably when content was updated.  The issue was the implementation of the `useContent` hook that contained a state controller.

image

### CODE REVIEW

File: app/editor/src/features/content/form/ContentForm.tsx [hunk 1]
```diff
@@ -193,7 +193,9 @@ export const ContentForm: React.FC<IContentFormProps> = ({
             if (contentId === +id) fetchContent(+id);
           });
         })
-        .catch((error) => console.error(error));
+        .catch((error) => {
+          console.error('signalr', error);
```
Comment (Fosol) [inline app/editor/src/features/content/form/ContentForm.tsx:197]: @CelloLiu I am seeing a lot of errors come out of failed connections.

File: app/editor/src/store/hooks/editor/useContent.ts [hunk 2]
```diff
@@ -30,74 +30,79 @@ export const useContent = (props?: IContentProps): [IContentState, IContentContr
...
-    },
-    attach: async (id: number, path: string) => {
-      return (
-        await dispatch<IContentModel>('attach-content', () => api.attach(id, path), 'content')
-      ).data;
-    },
-    storeFilter: actions.storeFilter,
-    storeFilterAdvanced: actions.storeFilterAdvanced,
-  }).current;
+  const controller = React.useMemo(
```
Comment (Fosol) [inline app/editor/src/store/hooks/editor/useContent.ts:33]: `useRef` as causing the issue.  I think I originally attempted to eliminate `useMemo` due to infinite state changes at some point in the past.

File: app/editor/src/store/hooks/editor/useWorkOrders.ts [hunk 3]
```diff
@@ -20,23 +20,26 @@ export const useWorkOrders = (): [any, IWorkOrderController] => {
...
-      return response;
-    },
-    transcribe: async (content: IContentModel) => {
-      return await dispatch('transcribe-content', () => api.transcribe(content));
-    },
-    nlp: async (content: IContentModel) => {
-      return await dispatch('nlp-content', () => api.nlp(content));
-    },
-  }).current;
+  const controller = React.useMemo(
```
Comment (Fosol) [inline app/editor/src/store/hooks/editor/useWorkOrders.ts:23]: Fixed this one too

File: network/nginx/config/default.conf [hunk 4]
```diff
@@ -83,6 +83,8 @@ server {
     proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
     proxy_set_header X-NginX-Proxy true;
     proxy_ssl_session_reuse off;
+    proxy_set_header Upgrade $http_upgrade;
```
Comment (Fosol) [inline network/nginx/config/default.conf:86]: Resolving SignalR issue.

**Comment by Fosol** [comment]
@jdtoombs Take a look at this to understand why state wasn't being updated correctly.

File: app/editor/src/store/hooks/editor/useContent.ts (hunk 2 above)
Comment (jdtoombs) [reply app/editor/src/store/hooks/editor/useContent.ts:33]: Ahhh I was close to this at one point lol. Glad you got it figured out. Thanks! :)

**Review by jdtoombs — APPROVED** [review]

**Pull request merged**